Source/Initialization/WarpXInitData.cpp

```diff
@@ -647,7 +647,8 @@ WarpX::InitLevelData (int lev, Real /*time*/)
     }
 
     if (costs[lev]) {
-        for (int i : costs[lev]->IndexArray()) {
+        const auto iarr = costs[lev]->IndexArray();
```

Comment:
How about `const auto&` instead of `const auto`?

Reply:
Well, in principle I would agree, but it turns out that if I do this the Fujitsu compiler gives me an internal compiler error  💣 (this is an issue with the Fujitsu compiler that I have reported)! So, for this reason, I would prefer either to merge the PR with a copy instead of a reference, or to wait for Fujitsu to fix the issue with their compiler (I don't know when this will happen).

Reply:
Got it.